sw/device/tests/hmac_dif_test.c

```diff
@@ -148,143 +159,95 @@ static bool check_message_length(const dif_hmac_t *hmac,
 }
 
 /**
- * Kick off the HMAC (or SHA256) run. Return |true| if there are no errors.
+ * Kick off the HMAC (or SHA256) run. Return `true` if there are no errors.
  */
 static bool run_hmac(dif_hmac_t *hmac) {
-  switch (dif_hmac_run(hmac)) {
+  switch (dif_hmac_process(hmac)) {
     case kDifHmacOk:
       break;
     case kDifHmacBadArg:
-      LOG_ERROR("Invalid arguments encountered running HMAC.\r\n");
+      LOG_ERROR("Invalid arguments encountered running HMAC.");
       return false;
     default:
-      LOG_ERROR("Unknown error encountered running HMAC.\r\n");
+      LOG_ERROR("Unknown error encountered running HMAC.");
       return false;
   }
 
   return true;
 }
 
 /**
- * Spin until HMAC is done running and the digest is available. Return |true| if
- * there are no errors.
+ * Read the HMAC digest and compare it to the expected result. Return `true` if
+ * there are no errors and the digests match.
  */
-static bool wait_for_hmac_done(dif_hmac_t *hmac) {
+static bool check_digest(const dif_hmac_t *hmac,
+                         const uint32_t *expected_digest) {
+  dif_hmac_digest_t digest_result;
   bool hmac_done;
   do {
-    switch (dif_hmac_done(hmac, &hmac_done)) {
-      case kDifHmacOk:
+    hmac_done = true;
+
+    switch (dif_hmac_digest_read(hmac, &digest_result)) {
+      case kDifHmacDigestOk:
         break;
-      case kDifHmacBadArg:
-        LOG_ERROR(
-            "Invalid arguments encountered checking HMAC run status.\r\n");
+      case kDifHmacDigestBadArg:
+        LOG_ERROR("Invalid arguments encountered reading HMAC digest.");
         return false;
+      case kDifHmacDigestProcessing:
+        hmac_done = false;
       default:
-        LOG_ERROR("Unknown error encountered checking HMAC run status.\r\n");
+        LOG_ERROR("Unknown error encountered reading HMAC digest.");
         return false;
     }
   } while (!hmac_done);
 
-  return true;
-}
-
-/**
- * Read the HMAC digest and compare it to the expected result. Return |true| if
- * there are no errors and the digests match.
- */
-static bool check_digest(const dif_hmac_t *hmac,
-                         const uint32_t *expected_digest) {
-  dif_hmac_digest_t digest_result;
-  switch (dif_hmac_digest_read(hmac, &digest_result)) {
-    case kDifHmacOk:
-      break;
-    case kDifHmacBadArg:
-      LOG_ERROR("Invalid arguments encountered reading HMAC digest.\r\n");
-      return false;
-    default:
-      LOG_ERROR("Unknown error encountered reading HMAC digest.\r\n");
-      return false;
-  }
-
   bool matches = true;
   for (int i = 0; i < 8; ++i) {
     if (expected_digest[i] != digest_result.digest[i]) {
-      LOG_ERROR(
-          "Digest mismatch, expected: 0x%X at index [%i] but got: "
-          "0x%X.\r\n",
-          expected_digest[i], i, digest_result.digest[i]);
+      LOG_ERROR("Digest mismatch, expected: 0x%X at index [%i] but got: 0x%X.",
```

Comment:
I guess this could be directly turned into `CHECK`, and this function can become `void`. The same probably applies to other similar functions.

Please see the comment in `run_test`.